tests/integration/suite/test_workloads.py

```diff
@@ -372,3 +440,18 @@ def wait_for_service_cluserip_reset(client, name, timeout=30):
         if time.time() - start > timeout:
             raise Exception('Timeout waiting for workload service')
     return services.data[0]
+
+
+def wait_for_workload(client, workload_id, timeout=30):
+    """Wait for a given workload to be active."""
+    def _get_result():
+        try:
+            return client.by_id_workload(
+                workload_id)["deploymentStatus"]["conditions"][-1]["status"]
```

Comment:
This should check that the `state` is active, just grabbing the last condition could change meaning at any time.

Reply:
Agreed. Done.